packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -224,35 +228,42 @@ class AndroidCameraCameraX extends CameraPlatform {
     // Start listening for device orientation changes preceding camera creation.
     startListeningForDeviceOrientationChange(
         cameraIsFrontFacing, cameraDescription.sensorOrientation);
+    // Determine ResolutionSelector and QualitySelector based on preset for
+    // camera UseCases.
+    final ResolutionSelector? presetResolutionSelector =
+        _getResolutionSelectorFromPreset(resolutionPreset);
+    final QualitySelector? presetQualitySelector =
+        _getQualitySelectorFromPreset(resolutionPreset);
 
     // Retrieve a fresh ProcessCameraProvider instance.
     processCameraProvider ??= await ProcessCameraProvider.getInstance();
     processCameraProvider!.unbindAll();
 
-    // TODO(camsim99): Implement resolution configuration for UseCases
-    // configured here. https://github.com/flutter/flutter/issues/120462
-
     // Configure Preview instance.
     final int targetRotation =
         _getTargetRotation(cameraDescription.sensorOrientation);
-    preview = createPreview(targetRotation);
+    preview = createPreview(
+        targetRotation: targetRotation,
+        resolutionSelector: presetResolutionSelector);
     final int flutterSurfaceTextureId = await preview!.setSurfaceProvider();
 
     // Configure ImageCapture instance.
-    imageCapture = createImageCapture(null);
+    imageCapture = createImageCapture(presetResolutionSelector);
+
+    // Configure ImageAnalysis instance.
+    // Defaults to YUV_420_888 image format.
+    imageAnalysis = createImageAnalysis(presetResolutionSelector);
 
     // Configure VideoCapture and Recorder instances.
-    // TODO(gmackall): Enable video capture resolution configuration in createRecorder().
-    recorder = createRecorder();
+    recorder = createRecorder(presetQualitySelector);
     videoCapture = await createVideoCapture(recorder!);
 
     // Bind configured UseCases to ProcessCameraProvider instance & mark Preview
     // instance as bound but not paused. Video capture is bound at first use
     // instead of here.
-    camera = await processCameraProvider!
-        .bindToLifecycle(cameraSelector!, <UseCase>[preview!, imageCapture!]);
-    await _updateLiveCameraState(flutterSurfaceTextureId);
-    cameraInfo = await camera!.getCameraInfo();
+    camera = await processCameraProvider!.bindToLifecycle(
+        cameraSelector!, <UseCase>[preview!, imageCapture!, imageAnalysis!]);
```

Comment:
These three use cases should be bound as soon as the camera is created and the video use case should be lazy loaded for best performance, according to the CameraX team. I must have forgotten this when implementing image analysis, so fixing here.